R/pkg/R/functions.R

```diff
@@ -163,3 +163,17 @@ setMethod("n", signature(x = "Column"),
           function(x) {
             count(x)
           })
+
+#' when
+#'
+#' Evaluates a list of conditions and returns one of multiple possible result expressions.
```

Comment:
The comments of `when` is same as that of `otherwise` in Scala. As you're suggesting, it would be better to explain the difference. Hmm, do you have any good idea?